src/main/java/com/microsoft/azure/kusto/kafka/connect/sink/GZIPFileWriter.java

```diff
@@ -40,37 +41,38 @@ public GZIPFileWriter(String basePath,
                           long fileThreshold,
                           Consumer<GZIPFileDescriptor> onRollCallback,
                           Supplier<String> getFilePath,
-                          long flushInterval) {
+                          long flushInterval,
+                          boolean flushImmediately) {
```

Comment:
to be honest. passing both flushInterval and flushImmediatly is kinda awkward.
It makes no sense to have both configured.
I would instead just pass in `flushIterval =0` & `fileThreshold = 0` to signify flushImmediately

Reply:
i don't think that we should do it like that
a zero flush interval will make the writer to invoke `flushByTimeImpl` function non-stop

Reply:
Of course you need to condition on it. I was just saying we dont need to change the interface to extend the capabilities here..